mule-extensions-api/src/main/java/org/mule/runtime/extension/api/model/ImmutableExtensionModel.java

```diff
@@ -102,6 +107,10 @@ public ImmutableExtensionModel(String name,
     checkModelArgument(vendor != null, "Extension Vendor cannot be null");
 
     checkModelArgument(xmlDslModel != null, "xmlDslModel cannot be null");
+    final boolean hasPrivilegedArtifacts = privilegedArtifacts != null && !privilegedArtifacts.isEmpty();
+    final boolean hasPrivilegedPackages = privilegedPackages != null && !privilegedPackages.isEmpty();
+    checkModelArgument(hasPrivilegedPackages == hasPrivilegedArtifacts,
```

Comment:
Add a model validator in `org.mule.runtime.extension.internal.loader.ExtensionModelFactory#extensionModelValidators` that checks for this consistency using the `ProblemsReporter`